anndata/_io/merge.py

```diff
@@ -0,0 +1,506 @@
+        for more.
+    merge
+        How elements not aligned to the axis being concatenated along are selected.
+        Currently implemented strategies include:
+
+        * `None`: No elements are kept.
+        * `"same"`: Elements that are the same in each of the objects.
+        * `"unique"`: Elements for which there is only one possible value.
+        * `"first"`: The first element seen at each from each position.
+        * `"only"`: Elements that show up in only one of the objects.
+    uns_merge
+        How the elements of `.uns` are selected. Uses the same set of strategies as
+        the `merge` argument, except applied recursively.
+    label
+        Column in axis annotation (i.e. `.obs` or `.var`) to place batch information in.
+        If it's None, no column is added.
+    keys
+        Names for each object being added. These values are used for column values for
+        `label` or appended to the index if `index_unique` is not `None`. Defaults to
+        incrementing integer labels.
+    index_unique
+        Whether to make the index unique by using the keys. If provided, this
+        is the delimeter between "{orig_idx}{index_unique}{key}". When `None`,
+        the original indices are kept.
+    fill_value
+        When `join="outer"`, this is the value that will be used to fill the introduced
+        indices. By default, sparse arrays are padded with zeros, while dense arrays and
+        DataFrames are padded with missing values.
+    pairwise
+        Whether pairwise elements along the concatenated dimension should be included.
+        This is False by default, since the resulting arrays are often not meaningful.
+
+    Notes
+    -----
+
+    .. warning::
+
+        If you use `join='outer'` this fills 0s for sparse data when
+        variables are absent in a batch. Use this with care. Dense data is
+        filled with `NaN`.
+    """
+    # Argument normalization
+    merge = resolve_merge_strategy(merge)
+    uns_merge = resolve_merge_strategy(uns_merge)
+
+    if isinstance(in_files, Mapping):
+        raise NotImplementedError("Not implemented just give path list")
+        # if keys is not None:
+        #     raise TypeError(
+        #         "Cannot specify categories in both mapping keys and using `keys`. "
+        #         "Only specify this once."
+        #     )
+        # keys, in_files = list(in_files.keys()), list(in_files.values())
+    else:
+        in_files = list(in_files)
+
+    if join != "inner":
+        raise NotImplementedError("Only inner is implemented")
+
+    # TODO: Generalize this to more than zarr
+    groups = [zarr.open(store=p) for p in in_files]
+    output_group = zarr.open(store=out_file)
+    assert len(groups) > 1
+
+    # var or obs
+    _, dim = _resolve_dim(axis=axis)
+    alt_axis, alt_dim = _resolve_dim(axis=1 - axis)
+
+    # TODO: This is just a temporary assertion
```

Comment:
Why is this temporary and when would it be removed?

Right now, it looks like this is loading in the obs_names/ var_names multiple times, but it would be nice to load them just once.